layouts/partials/sidebar.html

```diff
@@ -0,0 +1,52 @@
+<nav class="collapse srcf-links" id="srcf-docs-nav">
+  {{- $url := split .Permalink "/" -}}
+  {{- $page_slug := index $url (sub (len $url) 2) -}}
+
+    <ul class="list-unstyled mb-0 py-3 pt-md-1">
+    {{- range $section := .Site.Data.sidebar -}}
+    {{- $section_slug := $section.category | urlize -}}
+    {{- $is_active_section := eq $.Type $section_slug -}}
+
+    {{ if $is_active_section }}
+    
+    {{- range $group := $section.groups -}}
+    {{- $group_slug := $group.title | urlize -}}
+    {{- $is_active_group := eq $.Page.Params.group $group_slug -}}
+
+    <li class="mb-1">
+    {{- if $group.pages }}
+      <button class="btn d-inline-flex align-items-center rounded {{ if not $is_active_group }} collapsed {{ end }}" data-bs-toggle="collapse" data-bs-target="#{{ $group_slug }}-collapse" aria-expanded="{{ $is_active_group }}"{{ if $is_active_group }} aria-current="true"{{ end }}>
+        {{ $group.title }}
+      </button>
+
+      <div class="collapse{{ if $is_active_group }} show{{ end }}" id="{{ $group_slug }}-collapse">
```

Comment:
This has been discussed a fair bit already, but noting here for completeness: it would be nice to not have to depend on JavaScript for the sidebar.  For your consideration, here's a minimal but functional alternative using the native `<details>` HTML element:

```html
<details{{ if $is_active_group }} open{{ end }}>
  <summary>{{ $group.title }}</summary>
  <ul>...</ul>
</details>
```

```css
#srcf-docs-nav summary {
  padding-left: 1.75rem;
}
```

Reply:
I've chosen instead to remove the dropdowns entirely and add some padding on the left of the sub items to distinguish them. Note we still need JS for the collapse sidebar as a whole as otherwise that would make navigating on mobile really difficult.